pkg/kube/client.go

```diff
@@ -369,13 +370,55 @@ func batchPerform(infos ResourceList, fn func(*resource.Info) error, errs chan<-
 			kind = currentKind
 		}
 		wg.Add(1)
-		go func(i *resource.Info) {
-			errs <- fn(i)
+		go func(info *resource.Info) {
+			// try 3 times total
+			tries := 3
+			rand.Seed(time.Now().UTC().UnixNano())
+
+			fnErr := fn(info)
+			for counter := 1; counter < tries; counter++ {
+				// break out if this is not a Resource Quota Conflict of if the resource itself
+				// is a resourcequota
+				if fnErr == nil ||
+					kind == "resourcequotas" ||
+					!isResourceQuotaConflict(fnErr) {
+					break
+				}
+
+				// Increasing wait: 10-19ms, 40-59ms
+				waitTime := time.Duration(counter*counter*10+rand.Intn(counter*10)) * time.Millisecond
+				time.Sleep(waitTime)
```

Comment:
You can use `wait.Pull` from `k8s.io/apimachinery/pkg/util/wait`, not `time.Sleep`

Reply:
wait.Poll polls at regular intervals though and waits for a condition. I'll think about this, but I'd have to create another (nested) function with all the logic and pass that as a function variable?